plugins/modules/nd_site.py

```diff
@@ -0,0 +1,249 @@
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["site"]],
+            ["state", "present", ["site", "apic_username", "apic_password", "url", "site_type"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    apic_username = nd.params.get("apic_username")
+    apic_password = nd.params.get("apic_password")
+    apic_inband_epg = validate_not_none(nd.params.get("apic_inband_epg"))
+    apic_login_domain = validate_not_none(nd.params.get("apic_login_domain"))
+    site = nd.params.get("site")
+    url = nd.params.get("url")
+    site_type = nd.params.get("site_type")
+    latitude = validate_not_none(nd.params.get("latitude"))
+    longitude = validate_not_none(nd.params.get("longitude"))
+    security_domains = nd.params.get("security_domains")
+    state = nd.params.get("state")
+
+    path = "/api/config/class/v2/sites/"
+
+    # Query for existing object(s)
+    if site:
+        site_info = next((site_dict.get('name') for site_dict in nd.query_obj(path) if site_dict.get('name') == site), None)
+        if site_info:
+            path = "/api/config/dn/v2/sites/{0}".format(site)
+            nd.existing = nd.query_obj(path)
+    else:
+        nd.existing = nd.query_obj(path)
+
+    if state == "query":
+        pass
+    elif state == "absent":
+        nd.previous = nd.existing
+        if nd.existing:
+            if module.check_mode:
+                nd.existing = {}
+            else:
+                rm_path = "/api/config/v2/deletesite/"
+                rm_payload = {
+                    "name": site,
+                    "aci": {
+                        "userName": apic_username,
+                        "password": apic_password,
+                        "loginDomain": apic_login_domain,
+                    }
+                }
+                rm_resp = nd.request(rm_path, method="POST", data=rm_payload)
+                if rm_resp["response"] is not None:
+                    nd.existing = {}
+    elif state == "present":
+        nd.previous = nd.existing
+        add_path = "/api/config/v2"
+        if nd.existing:
+            payload = nd.existing
+        else:
+            payload = {
+                "name": site,
+                "url": url,
+                "siteType": site_type,
+                "aci": {
+                    "userName": apic_username,
+                    "password": apic_password,
+                    "loginDomain": apic_login_domain,
+                    "inbandEpgDN": apic_inband_epg,
+                },
+                "latitude": latitude,
+                "longitude": longitude,
+                "securityDomains": security_domains if security_domains is not None else [],
+            }
+
+        nd.sanitize(payload, collate=True)
+
+        if nd.existing:
+            if nd.check_changed():
```

Comment:
Should we combine the two if statements to single line?
```suggestion
        if nd.existing and nd.check_changed():
```